### PR TITLE
[Contracts] Add types to private properties

### PR DESCRIPTION
**Pull request by derrabus**
| Q             | A
| ------------- | ---
| Branch?       | 6.0
| Bug fix?      | no
| New feature?  | no
| Deprecations? | no
| Tickets       | N/A
| License       | MIT
| Doc PR        | N/A

### CODE REVIEW

File: src/Symfony/Contracts/Translation/TranslatorTrait.php [hunk 1]
```diff
@@ -35,7 +35,7 @@ public function setLocale(string $locale)
      */
     public function getLocale(): string
     {
-        return $this->locale ?: (class_exists(\Locale::class) ? \Locale::getDefault() : 'en');
+        return $this->locale ?? (class_exists(\Locale::class) ? \Locale::getDefault() : 'en');
```
Comment (nicolas-grekas) [inline src/Symfony/Contracts/Translation/TranslatorTrait.php]: this is going to change the outcome for the empty string

Comment (stof) [reply src/Symfony/Contracts/Translation/TranslatorTrait.php]: AFAIR, the component forbids using the empty string as locale. Maybe the setter should forbid the empty string here too.

Comment (derrabus) [reply src/Symfony/Contracts/Translation/TranslatorTrait.php]: I've initialized the property with `null` for now to keep things simple. If we want to disallow the empty string as @stof suggested, we would need to build a deprecation layer in constracts v2 first.

Comment (nicolas-grekas) [reply src/Symfony/Contracts/Translation/TranslatorTrait.php]: Actually not, see https://github.com/symfony/symfony/pull/42128...

**Review by nicolas-grekas — APPROVED** [review]

**Pull request merged**